x-pack/plugins/security_solution/public/detection_engine/rule_management/api/hooks/use_bulk_action_mutation.ts

```diff
@@ -18,40 +18,40 @@ import { DETECTION_ENGINE_RULES_BULK_ACTION } from '../../../../../common/consta
 export const BULK_ACTION_MUTATION_KEY = ['POST', DETECTION_ENGINE_RULES_BULK_ACTION];
 
 export const useBulkActionMutation = (
-  options?: UseMutationOptions<BulkActionResponse, Error, BulkActionProps>
+  options?: UseMutationOptions<BulkActionResponse, Error, BulkAction>
 ) => {
   const invalidateFindRulesQuery = useInvalidateFindRulesQuery();
   const invalidateFetchRuleByIdQuery = useInvalidateFetchRuleByIdQuery();
   const invalidateFetchTagsQuery = useInvalidateFetchTagsQuery();
   const invalidateFetchPrebuiltRulesStatusQuery = useInvalidateFetchPrebuiltRulesStatusQuery();
   const updateRulesCache = useUpdateRulesCache();
 
-  return useMutation<BulkActionResponse, Error, BulkActionProps>(
-    (action: BulkActionProps) => performBulkAction(action),
+  return useMutation<BulkActionResponse, Error, BulkAction>(
+    (bulkAction: BulkAction) => performBulkAction({ bulkAction }),
```

Comment:
Let's add back all the arguments accepted by the `performBulkAction` method. 

Hooks located in `api/hooks` have to be generic. They are designed as an abstraction layer above the API client methods matching them one to one. And these hooks' only responsibility is to define query keys and invalidate other related queries. So hook's interface should match its underlying API method exactly.

We will prohibit direct API usage in the future so that these hooks will become the only API "client" facing users. See https://github.com/elastic/kibana/issues/143923